pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go

```diff
@@ -0,0 +1,162 @@
+package agentpoller
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+// GuestOsInfo is the response from 'guest-get-osinfo'
+type GuestOsInfo struct {
+	Name          string `json:"name"`
+	KernelRelease string `json:"kernel-release"`
+	Version       string `json:"version"`
+	PrettyName    string `json:"pretty-name"`
+	VersionId     string `json:"version-id"`
+	KernelVersion string `json:"kernel-version"`
+	Machine       string `json:"machine"`
+	Id            string `json:"id"`
+}
+
+// Interface for json unmarshalling
+type Interface struct {
+	MAC  string `json:"hardware-address"`
+	IPs  []IP   `json:"ip-addresses"`
+	Name string `json:"name"`
+}
+
+// IP for json unmarshalling
+type IP struct {
+	IP     string `json:"ip-address"`
+	Type   string `json:"ip-address-type"`
+	Prefix int    `json:"prefix"`
+}
+
+// stripAgentResponse use regex to strip the wrapping item and returns the
+// embeded object.
+// It is a workaround so the amount of copy paste code is limited
+func stripAgentResponse(agentReply string) string {
+	re := regexp.MustCompile(`{\s*\"return\":\s*([{\[][\s\S]*[}\]])\s*}`)
+	return re.FindStringSubmatch(agentReply)[1]
+}
+
+// parseGuestOSInfo parse agent reply string, extract guest os info
+// and converts the response to API domain guest os info
+func parseGuestOSInfo(agentReply string) api.GuestOSInfo {
+	guestOSInfo := GuestOsInfo{}
+	response := stripAgentResponse(agentReply)
+
+	err := json.Unmarshal([]byte(response), &guestOSInfo)
+	if err != nil {
+		log.Log.Errorf("Cannot parse guestosinfo reply: %s", err.Error())
+		return api.GuestOSInfo{}
+	}
+
+	resultInfo := api.GuestOSInfo{
+		Name:          guestOSInfo.Name,
+		KernelRelease: guestOSInfo.KernelRelease,
+		Version:       guestOSInfo.Version,
+		PrettyName:    guestOSInfo.PrettyName,
+		VersionId:     guestOSInfo.VersionId,
+		KernelVersion: guestOSInfo.KernelVersion,
+		Machine:       guestOSInfo.Machine,
+		Id:            guestOSInfo.Id,
+	}
+
+	return resultInfo
+}
+
+// parseInterfaces parse agent reply string, extract network interfaces
+// and converts the response to API domain list of interfaces
+func parseInterfaces(agentReply string) []api.InterfaceStatus {
+	interfaces := []Interface{}
+	response := stripAgentResponse(agentReply)
+
+	err := json.Unmarshal([]byte(response), &interfaces)
+	if err != nil {
+		log.Log.Errorf("Cannot parse interface reply: %s", err.Error())
```

Comment:
If the error was returned, this would enable better unit testing by being able to check for error conditions.